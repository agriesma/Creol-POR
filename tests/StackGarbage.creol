class Test
begin
  op init == skip
  op run == var r: Int; m(; r)
  op m(out r: Int) == begin m(; r); r := r + 1 end [] r := 1
end
