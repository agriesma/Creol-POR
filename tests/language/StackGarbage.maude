load creol-interpreter
mod PROGRAM is
protecting CREOL-SIMULATOR .
op init : -> Configuration [ctor] .
eq init =
< "Test" : Cl | Inh: noInh, Par: noVid, Att: noSubst, Mtds: 
  < "init" : Mtdname | Param: noVid, Latt: noSubst, Code: skip ; return ( emp ) > *
  < "run" : Mtdname | Param: noVid, Latt: "label:0" |-> null , "r" |-> null, Code: ( "label:0" ! "this" . "m" (emp ) )  ; ( "label:0" ? ( "r" ) )  ; return ( emp ) > *
  < "m" : Mtdname | Param: noVid, Latt: "r" |-> null , "label:0" |-> null, Code: ( ( "label:0" ! "this" . "m" (emp ) )  ; ( "label:0" ? ( "r" ) )  ; "r" ::= "+" ( "r" # int(1) ) ||| "r" ::= int(1) ) ; return ( "r" ) >, Ocnt: 0 >

.
endm
