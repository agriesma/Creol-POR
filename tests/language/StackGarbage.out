==========================================
reduce in PROGRAM : init .
result Class: 
< "Test" : Cl | Inh: noInh, Par: noVid, Att: noSubst, Mtds:
  < "init" : Mtdname | Param: noVid, Latt: noSubst, Code: skip ; return(emp) >
    *
  < "m" : Mtdname | Param: noVid, Latt: "label:0" |-> null, "r" |-> null, Code:
    ("r" ::= int(1) ||| "label:0" ! "this" . "m"(emp) ; "label:0" ?("r") ; "r"
    ::= "+"("r" # int(1))) ; return("r") > *
  < "run" : Mtdname | Param: noVid, Latt: "label:0" |-> null, "r" |-> null,
    Code: "label:0" ! "this" . "m"(emp) ; "label:0" ?("r") ; return(emp) >,
  Ocnt: 0 >
Bye.
